### ISSUE TITLE
Make HyperLink rather an optional attribute of other widget. Remove HyperLinkSegment

### ISSUE THREAD

**Issue by EugeneSindambiwe** [fyne-io/fyne#2656]
Hi everyone,
(1) Obviously images need hyperlink as well. 
(2) Also I'd conceptually argue that a hyperlink isn't a widget. 
(3) Finally, I can anticipate technical challenges to cover (1) coherently in the current design

Could you please consider adding (compatible) or replacing (not compatible) HyperLink as an attribute of any widget.
Doing so would cover the same scope as Hyperlink in HTML, i.e. ability to navigate from any UI element to any resource with a URL. No HyperLinkSegment would be needed afterwards.

Thanks

**Comment by EugeneSindambiwe** [comment]
Basically, I consider a non-input text field, text fragment, image to be special cases of non-hidden and non-editable UI elements. With a hyperlink attribute, it could be possible to make visible to the end user that the UI element is navigable like an href tag in HTML. 
Thanks for the hint. I have to confess, a Tapped handler of course works and is still needed. Merely a Hyperlink would be visually recognizable as such by still keeping its narrow semantic, i.e. opening a URL. Either way I am fine how you deal the proposal.

**Comment by andydotxyz** [comment]
> Basically, I consider a non-input text field, text fragment, image to be special cases

That is probably true. A hyperlink could technically be an image as well - however I am confident that this should not mean that all widgets can become hyperlinks.
The `Hyperlink` provides the non-input text and the `RichText` feature `HyperLinkSegment` covers the fragment. Perhaps the right approach is to add an optional image to the `HyperLink` widget, that way I think we would satisfy your request, just in a different way?

**Comment by Jacalz** [comment]
Wouldn't it be better to add a `Link` field to the `widget.Icon` rather than adding an image to each hyperlink?

**Comment by EugeneSindambiwe** [comment]
I agree with the approach to qualify a subset of hyperlinkable widgets with image and icon included. We don't have to add unnecessary complexity. Thanks

**Comment by Jacalz** [comment]
I was not suggesting adding it to `canvas.Image`. That's just a canvas primitive. If developers want functionality on top of that, they can implement it rather easily as discussed above.

**Comment by andydotxyz** [comment]
> Wouldn't it be better to add a `Link` field to the `widget.Icon` rather than adding an image to each hyperlink?

That is not what I was proposing.
I was thinking an optional Image field like icon on Button.

**Comment by Jacalz** [comment]
> > Wouldn't it be better to add a `Link` field to the `widget.Icon` rather than adding an image to each hyperlink?
> 
> That is not what I was proposing. I was thinking an optional Image field like icon on Button.

Yeah, I know. I should have clarified a bit more clearly what I meant. I understand that each hyperlink won't need to display an image but I mean that it is easier to make an icon tappable than duplicating a lot of image rendering logic just tomake sure that hypelinks can show images if desired.

**Comment by andydotxyz** [comment]
> it is easier to make an icon tappable than duplicating a lot of image rendering logic

I guess I don't understand what rendering logic would be replicated. Though the more I think about it the more I am uncertain of the expected approach. Should images be a standard size like normal hyperlink? (unlikely) Or should they be "original" size (causing variance in visible size based on scale factor?
Neither seems ideal.

**Comment by andydotxyz** [comment]
Edited title to better match the scope we seem to have settled on.

**Comment by Jacalz** [comment]
What I mean is that it is a lot less lines of code to add if we just add a link option to the icon than to add the ability for the hyperlink widget to render images as well. Adding images to the hypelink widget also seems kind of strange to me as you suddenly make it either render text or render an image.

**Comment by andydotxyz** [comment]
Yes it may be less code.
However as we need to indicate the different modality (hyperlinks have underlines basically) then it would change the geometry and layout of images when link is turned on.
I don't think that this is the trivial change alternative that you presented.

**Comment by andydotxyz** [comment]
This may be possible if following the rich text expectation that such inline images in larger documents appear at half original pixel size so compromising exactly pixel density to get consistent sizing.

If we take this approach it will need to be new code in `Hyperlink` as `Icon` does not work this way.
